File: gto-support-androidx-room/src/testFixtures/java/org/ccci/gto/android/common/androidx/room/TestRoomDatabaseCoroutines.java
package org.ccci.gto.android.common.androidx.room;

import android.annotation.SuppressLint;

import androidx.room.RoomDatabase;

import kotlinx.coroutines.CoroutineScope;

public class TestRoomDatabaseCoroutines {
    @SuppressLint("RestrictedApi")
    public static void setCoroutineScope(RoomDatabase db, CoroutineScope scope) {
        db.getBackingFieldMap().put(RoomDatabase_CoroutinesKt.FIELD_COROUTINE_SCOPE, scope);
    }
}

File: gto-support-androidx-room/src/testFixtures/kotlin/org/ccci/gto/android/common/androidx/room/TestRoomDatabaseCoroutines.kt
package org.ccci.gto.android.common.androidx.room

import androidx.room.RoomDatabase
import kotlinx.coroutines.CoroutineScope

object TestRoomDatabaseCoroutines {
    @Deprecated(
        "Since v4.4.0, define a custom CoroutineContext when creating the RoomDatabase instead.",
        level = DeprecationLevel.ERROR
    )
    fun setCoroutineScope(db: RoomDatabase, scope: CoroutineScope): Nothing = error("This is no longer supported")
}
